### PR TITLE
[FIXED JENKINS-15057] Add dependency resolution to manually uploaded plugins.

### PR DESCRIPTION
**Pull request by slide**
Uses the manifest file to check plugin dependencies and automatically download them if possible.

### CODE REVIEW

**Comment by kzantow** [comment]
👍 AFAICT

File: test/src/test/resources/plugins/upload-test-update-center.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
+updateCenter.post(
+{"connectionCheckUrl":"http://www.google.com/", // has to be here because updateDirectly sniffs only very limited formats
```
Comment (daniel-beck) [inline test/src/test/resources/plugins/upload-test-update-center.json]: I don't think this comment works.

Comment (slide) [inline test/src/test/resources/plugins/upload-test-update-center.json]: It's what is in there for another test, but I will remove it.

**Comment by oleg-nenashev** [comment]
:+1:

**Comment by slide** [comment]
Can someone who has access merge this?

**Comment by oleg-nenashev** [comment]
@daniel-beck @olivergondza 
Do we still accept changes into 2.0?
Maybe makes sense to delay the fix merge till 2.1.
@slide WDYT?

**Comment by slide** [comment]
I forgot to mention that this fixes JENKINS-15057

**Comment by daniel-beck** [comment]
@oleg-nenashev 

> Do we still accept changes into 2.0?
> Maybe makes sense to delay the fix merge till 2.1.

`master` is towards 2.1. 2.0 will be released from the `2.0` branch. See my email to the dev list.

**Comment by slide** [comment]
So, this could be merged then since master is 2.1

**Comment by slide** [comment]
Thanks!

**Pull request merged**